sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -439,25 +443,29 @@ public static <T> SerializableFunction<Row, T> getFromRowFunction(Class<T> clazz
 
   /** Returns a function mapping encoded AVRO {@link GenericRecord}s to Beam {@link Row}s. */
   public static SimpleFunction<byte[], Row> getAvroBytesToRowFunction(Schema beamSchema) {
```

Comment:
Maybe it'd make sense here to allow users to pass in a factory for `DatumReader` and `DatumWriter` instances.  Then there could be this implementation, or others if users have their own.  This is similar to how [AvroIO.readFiles](https://beam.apache.org/releases/javadoc/2.39.0/org/apache/beam/sdk/io/AvroIO.ReadFiles.html#withDatumReaderFactory-org.apache.beam.sdk.io.AvroSource.DatumReaderFactory-) works.